http-client/src/main/java/io/airlift/http/client/HttpClientConfig.java

```diff
@@ -649,6 +650,19 @@ public HttpClientConfig setTcpKeepAliveIdleTime(Duration tcpKeepAliveIdleTime)
         return this;
     }
 
+    @NotNull
+    public Optional<Integer> getIpTos()
```

Comment:
Add validation for the TOS range:
```java
@Min(0)
@Max(255)
```
This matches the validation in the JDK `sun.nio.ch.Net#setSocketOption`